src/segment-loader.js

```diff
@@ -624,6 +661,8 @@ export default class SegmentLoader extends videojs.EventTarget {
    * @private
    */
   fillBuffer_() {
+    // TODO since the source buffer maintains a queue, and we shouldn't call this function
+    // except when we're ready for the next segment, this check can most likely be removed
     if (this.sourceUpdater_.updating()) {
```

Comment:
This exists for a very important reason...  the particulars of while elude me now. 🎗